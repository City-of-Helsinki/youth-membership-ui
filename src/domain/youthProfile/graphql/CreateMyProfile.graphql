mutation CreateMyProfile($input: CreateMyProfileMutationInput!) {
  createMyProfile(input: $input) {
    profile {
      id
    }
  }
}
