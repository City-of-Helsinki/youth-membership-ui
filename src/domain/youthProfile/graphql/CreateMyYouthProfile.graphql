mutation CreateMyYouthProfile($input: CreateMyYouthProfileMutationInput!) {
  createMyYouthProfile(input: $input) {
    youthProfile {
      birthDate
    }
  }
}
