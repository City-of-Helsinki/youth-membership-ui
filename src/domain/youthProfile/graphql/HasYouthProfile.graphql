query HasYouthProfile {
  myYouthProfile {
    id
    membershipStatus
  }
}
