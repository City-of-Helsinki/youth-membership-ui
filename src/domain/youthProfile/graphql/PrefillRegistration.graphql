query PrefillRegistartion {
  myProfile {
    firstName
    lastName
    language
    primaryPhone {
      phone
      id
    }
    primaryAddress {
      address
      postalCode
      city
      countryCode
      id
      primary
      addressType
    }
    addresses {
      edges {
        node {
          primary
          id
          address
          postalCode
          city
          countryCode
          addressType
        }
      }
    }
    primaryEmail {
      email
    }
  }
}
