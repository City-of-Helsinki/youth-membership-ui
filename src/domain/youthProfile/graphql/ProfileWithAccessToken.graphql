query ProfileWithAccessToken($token: UUID!) {
  profileWithAccessToken(token: $token) {
    firstName
    lastName
    language
    id
    primaryAddress {
      address
      postalCode
      city
      countryCode
      id
      addressType
      primary
    }
    addresses {
      edges {
        node {
          primary
          id
          address
          postalCode
          city
          countryCode
          addressType
        }
      }
    }
    primaryEmail {
      email
      id
    }
    primaryPhone {
      phone
      id
    }
  }
}
