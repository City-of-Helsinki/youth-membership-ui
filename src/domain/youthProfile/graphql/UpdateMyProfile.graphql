mutation UpdateMyProfile($input: UpdateMyProfileMutationInput!) {
  updateMyProfile(input: $input) {
    profile {
      youthProfile {
        approverEmail
      }
    }
  }
}
