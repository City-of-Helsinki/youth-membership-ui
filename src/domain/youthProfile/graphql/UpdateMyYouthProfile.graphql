mutation UpdateMyYouthProfile($input: UpdateMyYouthProfileMutationInput!) {
    updateMyYouthProfile(input: $input) {
        youthProfile {
            approverEmail
        }
    }
}
