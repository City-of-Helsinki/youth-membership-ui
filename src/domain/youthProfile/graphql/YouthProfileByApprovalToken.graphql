query YouthProfileByApprovalToken($token: String!) {
  youthProfileByApprovalToken(token: $token) {
    birthDate
    schoolName
    schoolClass
    languageAtHome
    photoUsageApproved
    approverFirstName
    approverLastName
    approverEmail
    approverPhone
    membershipNumber
    additionalContactPersons {
      edges {
        node {
          id
          firstName
          lastName
          phone
          email
        }
      }
    }
  }
}
