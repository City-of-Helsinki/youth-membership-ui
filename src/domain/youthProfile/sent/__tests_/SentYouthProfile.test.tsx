import React from 'react';
import { MemoryRouter } from 'react-router';
import { loader } from 'graphql.macro';
import toJson from 'enzyme-to-json';

import ViewYouthProfile from '../SentYouthProfile';
import {
  mountWithProviders,
  updateWrapper,
} from '../../../../common/test/testUtils';

const APPROVER_EMAIL = loader('../../graphql/ApproverEmail.graphql');

const mocks = [
  {
    request: {
      query: APPROVER_EMAIL,
      variables: {},
    },
    result: {
      data: {
        myYouthProfile: {
          approverEmail: 'ville.vanhempi@test.fi',
          __typename: 'YouthProfileType',
        },
      },
      loading: false,
    },
  },
];

const getWrapper = () => {
  return mountWithProviders(
    <MemoryRouter>
      <ViewYouthProfile />
    </MemoryRouter>,
    mocks
  );
};

test('match snapshot', async () => {
  const wrapper = getWrapper();
  await updateWrapper(wrapper);
  const hostingBox = wrapper.find('.hostingBox');
  expect(toJson(hostingBox)).toMatchSnapshot();
});

test('renders view with approver email', async () => {
  const wrapper = getWrapper();
  await updateWrapper(wrapper);

  const helpText = wrapper.find('.helpText').text();
  expect(helpText).toEqual(
    'Olet lähettänyt jäsenyyden hyväksyttäväksi osoitteeseen ville.vanhempi@test.fi.'
  );
});
