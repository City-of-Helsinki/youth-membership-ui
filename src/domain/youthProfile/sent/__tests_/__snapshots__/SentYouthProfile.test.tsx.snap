// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div>
  <div
    class="hostingBox"
  >
    <h1
      class="text h1 "
    >
      Hakemus on lähetetty huoltajallesi!
    </h1>
    <p
      class="helpText"
    >
      Olet lähettänyt jäsenyyden hyväksyttäväksi osoitteeseen
       
      .
    </p>
    <button
      class="Button-module_button__1msFE button_hds-button__2A0je Button-module_primary__2LfKB button_hds-button--primary__2NVvO button"
      type="button"
    >
      <span
        class="Button-module_label__a4np1 button_hds-button__label__2EQa-"
      >
        Lähetä uusi pyyntö huoltajallle
      </span>
    </button>
    <br />
    <a
      class="hds-button hds-button--secondary button"
      href="/membership-details"
    >
      <span
        class="hds-button__label"
      >
        Näytä hakemuksen tiedot
      </span>
      
    </a>
  </div>
</div>
`;
